.github/workflows/pre-release.yml

```
@@ -0,0 +1,100 @@
+
+      - name: Generate release notes
+        run: |
+          mkdir -p releases
+          # Generate release notes using your script
+          python scripts/generate_evolution_notes.py ${{ steps.version.outputs.version }} --output-dir releases
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Verify release notes file exists
+        id: verify_release_notes
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:7:50: Double quote to prevent globbing and word splitting

(shellcheck)